### PR TITLE
263221: Changed Child Beneficiary List to server side processing

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by lwyszomi** [comment]
@emord 
Added changes for:
Przemek issues:
https://manage.dimagi.com/default.asp?264585
https://manage.dimagi.com/default.asp?263221

Łukasz issues:
https://manage.dimagi.com/default.asp?263295
https://manage.dimagi.com/default.asp?264546
https://manage.dimagi.com/default.asp?264555

File: custom/icds_reports/reports/adhaar.py [hunk 1]
```diff
@@ -158,6 +161,7 @@ def get_adhaar_sector_data(domain, config, loc_level, location_id, show_test=Fal
     }
 
 
+@quickcache(['domain', 'config', 'loc_level', 'show_test'], timeout=30 * 60)
```
Comment (emord) [inline custom/icds_reports/reports/adhaar.py]: These quick cache's should be cleared when the aggregation task is finishes https://github.com/dimagi/commcare-hq/blob/master/custom/icds_reports/tasks.py#L106-L110

File: custom/icds_reports/reports/adolescent_girls.py [hunk 2]
```diff
@@ -119,7 +122,7 @@ def get_adolescent_girls_sector_data(domain, config, loc_level, location_id, sho
     chart_data['blue'] = sorted(chart_data['blue'])
 
     return {
-        "tooltips_data": tooltips_data,
+        "tooltips_data": dict(tooltips_data),
```
Comment (emord) [inline custom/icds_reports/reports/adolescent_girls.py]: Why does this need a cast

File: custom/icds_reports/views.py [hunk 3]
```diff
@@ -548,11 +548,25 @@ def get(self, request, *args, **kwargs):
             )
         elif step == 'beneficiary':
             if 'awc_id' in config:
+                start = request.GET.get('start', 0)
+                length = request.GET.get('length', 10)
+                draw = request.GET.get('draw', 0)
+
+                order_by_number_column = request.GET.get('order[0][column]')
```
Comment (emord) [inline custom/icds_reports/views.py]: Does this work? I've never seen something like this before and I don't see an example in django's docs

**Review by emord — CHANGES_REQUESTED** [review]
Build failure looks real

```
======================================================================
ERROR: custom.icds_reports.tests.reports.test_early_initiation_breastfeeding:TestEarlyInitiationBreastFeeding.test_sector_data
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/custom/icds_reports/tests/reports/test_early_initiation_breastfeeding.py", line 144, in test_sector_data
    loc_level='supervisor'
  File "/vendor/lib/python2.7/site-packages/quickcache/quickcache.py", line 40, in inner
    return helper(*args, **kwargs)
  File "/vendor/lib/python2.7/site-packages/quickcache/quickcache_helper.py", line 145, in __call__
    return self.call(*args, **kwargs)
  File "/vendor/lib/python2.7/site-packages/quickcache/quickcache_helper.py", line 55, in call
    key = self.get_cache_key(*args, **kwargs)
  File "/vendor/lib/python2.7/site-packages/quickcache/quickcache_helper.py", line 115, in get_cache_key
    callargs = inspect.getcallargs(self.fn, *args, **kwargs)
  File "/usr/local/lib/python2.7/inspect.py", line 988, in getcallargs
    'arguments' if num_required > 1 else 'argument', num_total))
TypeError: get_early_initiation_breastfeeding_data() takes at least 4 arguments (3 given)
----------------------------------------------------------------------
```

File: custom/icds_reports/reports/adolescent_girls.py (hunk 2 above)
Comment (lwyszomi) [reply custom/icds_reports/reports/adolescent_girls.py]: @emord because ```defaultdict``` with lambda function is not cachable by ```quickcache```.

**Comment by emord** [comment]
Looks like the tests pass now. Are you guys still working on my other two comments?

File: custom/icds_reports/views.py (hunk 3 above)
Comment (lwyszomi) [reply custom/icds_reports/views.py]: @emord Yes this works, This strange key is added by angular-datatable automatically.

File: custom/icds_reports/reports/adhaar.py (hunk 1 above)
Comment (emord) [reply custom/icds_reports/reports/adhaar.py]: (offlined)

File: custom/icds_reports/views.py (hunk 3 above)
Comment (emord) [reply custom/icds_reports/views.py]: Thanks. Didn't realize this was from angular

**Review by emord — APPROVED** [review]

**Pull request merged**